watertap/property_models/coagulation_prop_pack.py

```diff
@@ -0,0 +1,640 @@
+            if (p == 'Liq'):
+                return (b.dens_mass_phase[p] == (b.params.ref_dens_liq +
+                            b.params.dens_slope * b.mass_frac_phase_comp['Liq', 'TDS'] +
+                            b.params.dens_slope * b.mass_frac_phase_comp['Liq', 'TSS']) *
+                            (b.params.dens_param_A * b.temperature**2 + b.params.dens_param_B *
+                            b.temperature + b.params.dens_param_C) * (b.params.ref_pressure_correction +
+                            b.params.ref_pressure_slope * b.pressure) )
+            elif (p == 'Sol'):
+                return b.dens_mass_phase[p] == b.params.ref_dens_sol
+            else:
+                raise PropertyPackageError("Unsupported phase for CoagulationParameterData property package")
+        self.eq_dens_mass_phase = Constraint(self.params.phase_list, rule=rule_dens_mass_phase)
+
+
+    def _flow_vol_phase(self):
+        self.flow_vol_phase = Var(
+            self.params.phase_list,
+            initialize={'Liq': 1e-3, 'Sol': 1e-7},
+            bounds=(1e-16, None),
+            units=pyunits.m**3/pyunits.s,
+            doc="Volumetric flow rate")
+
+        def rule_flow_vol_phase(b, p):
+            if (p == 'Liq'):
+                return (b.flow_vol_phase[p]
+                    == (b.flow_mass_phase_comp['Liq', 'H2O'] +
+                            b.flow_mass_phase_comp['Liq', 'TDS'] +
+                                b.flow_mass_phase_comp['Liq', 'TSS'])
+                                    / b.dens_mass_phase[p])
+            elif (p == 'Sol'):
+                return (b.flow_vol_phase[p]== (b.flow_mass_phase_comp[p, 'Sludge'] / b.dens_mass_phase[p]) )
+            else:
+                raise PropertyPackageError("Unsupported phase for CoagulationParameterData property package")
+        self.eq_flow_vol_phase = Constraint(self.params.phase_list, rule=rule_flow_vol_phase)
+
+    def _conc_mass_phase_comp(self):
+        self.conc_mass_phase_comp = Var(
+            self.seawater_mass_frac_dict.keys(),
+            initialize=self.seawater_mass_conc_dict,
+            bounds=(1e-16, None),
+            units=pyunits.kg/pyunits.m**3,
+            doc="Mass concentration")
+
+        def rule_conc_mass_phase_comp(b, p, j):
+            return (b.conc_mass_phase_comp[p, j] ==
+                    b.mass_frac_phase_comp[p, j] * b.dens_mass_phase[p])
+        self.eq_conc_mass_phase_comp = Constraint(self.seawater_mass_frac_dict.keys(), rule=rule_conc_mass_phase_comp)
+
+    def _enth_flow(self):
+        # enthalpy flow expression for get_enthalpy_flow_terms method
+        temperature_ref = 273.15 * pyunits.K
+
+        def rule_enth_flow(b):  # enthalpy flow [J/s]
+            return (b.params.cp
+                    * sum(b.flow_mass_phase_comp[pair] for pair in b.flow_mass_phase_comp)
+                    * (b.temperature - temperature_ref))
+        self.enth_flow = Expression(rule=rule_enth_flow)
```

Comment:
I recently learned that IDAES control volumes expect enthalpy flow to be indexed by phase. An example of this is the [water_prop_pack](https://github.com/watertap-org/watertap/blob/add_initial_mvc/watertap/property_models/water_prop_pack.py) on the add_initial_mvc branch. I don't think this should hold up the PR, but if you use a control volume for your unit model and use the add_energy_balances, you'll need to make sure whatever is returned by the `get_enthalpy_flow_terms` method below is indexed by p.